### PR TITLE
Added datagram identifiers

### PR DESCRIPTION
**Pull request by DavidSchinazi**

### CODE REVIEW

File: draft-pauly-quic-datagram.md [hunk 1]
```diff
@@ -171,6 +182,14 @@ datagrams within a single QUIC connection, as the Stream ID does for
 STREAM frames. Demultiplexing datagram data is the responsibility of the
 application.
 
+## Datagram Identifiers {#datagram-id}
+
+Since several applications relying on datagrams have the need to identify which
+application-level flow a given datagram is a part of, DATAGRAM frames carry a
+datagram identifier. Applications that do not have a need for the identifier
+can use the value zero on their DATAGRAM frames and use the DATAGRAM_ID bit
+to omit sending the identifier over the wire.
```
Comment (tfpauly) [inline draft-pauly-quic-datagram.md]: Should we add text to say that if a mix of DATAGRAM+ID frames are used with DATAGRAM+NO_ID frames, that the non-ID frames are implicitly part of the "0" flow?

**Review by tfpauly — APPROVED** [review]
Looks good! Just one comment on a clarification I was thinking about.

File: draft-pauly-quic-datagram.md (hunk 1 above)
Comment (erickinnear) [reply draft-pauly-quic-datagram.md]: I had been thinking this as well, since that makes the application interface interaction really nice and clean.

**Review by erickinnear — APPROVED** [review]
Looks good to me! 
Might be worth wordsmithing the motivation for the Datagram ID a little bit more, but the construction seems sound.

**Pull request merged**